src/imitation/algorithms/sqil.py

```diff
@@ -0,0 +1,321 @@
+"""Soft Q Imitation Learning (SQIL).
+
+Trains a policy via DQN-style Q-learning,
+replacing half the buffer with expert demonstrations and adjusting the rewards.
+"""
+from typing import Any, Dict, Iterable, Optional, Tuple, Type, Union
+
+import numpy as np
+import torch as th
+import torch.nn.functional as F
+from stable_baselines3 import dqn
+from stable_baselines3.common import policies, vec_env
+from stable_baselines3.common.buffers import ReplayBuffer
```

Comment:
Nit: this codebase follows the Google Python Style Guide (except where it conflicts with PEP8, which takes precedence) and https://google.github.io/styleguide/pyguide.html#22-imports states:
> Use import statements for packages and modules only, not for individual classes or functions.

(We've tended to ignore this for tutorials as users may be less familiar with this style, but are more strict about adhering to it in the rest of the code.)